eras/conway/impl/src/Cardano/Ledger/Conway/PParams.hs

```diff
@@ -1,62 +1,229 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
 {-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
 {-# LANGUAGE DerivingVia #-}
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE GADTs #-}
 {-# LANGUAGE InstanceSigs #-}
+{-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE RecordWildCards #-}
 {-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
 {-# LANGUAGE TypeApplications #-}
 {-# LANGUAGE TypeFamilies #-}
 {-# LANGUAGE UndecidableInstances #-}
 {-# OPTIONS_GHC -Wno-orphans #-}
+{-# OPTIONS_GHC -Wno-unrecognised-pragmas #-}
 
 -- | This module contains the type of protocol parameters and EraPParams instance
 module Cardano.Ledger.Conway.PParams (
   BabbagePParams (..),
+  ConwayPParams (..),
   getLanguageView,
   LangDepView (..),
   encodeLangViews,
+  upgradeConwayPParams,
+  UpgradeConwayPParams (..),
 )
 where
 
 import Cardano.Ledger.Alonzo.PParams (OrdExUnits (..))
-import Cardano.Ledger.Alonzo.Scripts (ExUnits (..))
+import Cardano.Ledger.Alonzo.Scripts (CostModels, ExUnits (..), Prices (Prices), emptyCostModels)
+import Cardano.Ledger.Babbage (BabbageEra)
 import Cardano.Ledger.Babbage.Core
 import Cardano.Ledger.Babbage.PParams
+import Cardano.Ledger.BaseTypes (EpochNo (EpochNo), NonNegativeInterval, ProtVer (ProtVer), UnitInterval)
+import Cardano.Ledger.Binary
+import Cardano.Ledger.Binary.Coders
+import Cardano.Ledger.Coin (Coin (Coin))
+import Cardano.Ledger.Conway.Core
 import Cardano.Ledger.Conway.Era (ConwayEra)
+import Cardano.Ledger.Conway.Governance (ConwayGovernance)
 import Cardano.Ledger.Crypto
 import Cardano.Ledger.HKD (HKD, HKDFunctor (..))
-import Data.Coerce
+import Control.DeepSeq (NFData)
+import Data.Aeson hiding (Encoding, decode, encode)
+import qualified Data.Aeson as Aeson
+import Data.Default.Class (Default (def))
+import Data.Functor.Identity (Identity)
+import qualified Data.Map.Strict as Map
+import Data.Maybe.Strict (StrictMaybe (..), isSNothing)
 import Data.Proxy
+import GHC.Generics (Generic)
 import Lens.Micro
+import NoThunks.Class (NoThunks)
+import Numeric.Natural (Natural)
+
+-- | Conway Protocol parameters. Ways in which parameters have changed from Babbage:
+-- addition of @drepDeposit@
```

Comment:
Seems like we should either mention all the new ones here or none? it's a little confusing to only mention one , no?

Reply:
Thanks for the catch!